common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -303,6 +293,42 @@ private void reportLeak() {
         }
     }
 
+    /**
+     * Throws AssertionError if there's been any leaks.
+     * The current thread should have already been synchronized with any other threads that were responsible for
+     * releasing tracked objects. Any tracked object which has not been disposed (whether or not it is reachable
+     * or has been collected by the GC) will be considered a leak.
+     */
+    public void assertAllResourcesDisposed() {
+        // All hetherto unreported leaks, whether they've already been queued by the GC or not, will be here.
+        for (DefaultResourceLeak<?> leak : allLeaks.keySet()) {
+            leak.dispose(); // will always return true
+            String records = leak.toString();
+            if (reportedLeaks.putIfAbsent(records, Boolean.TRUE) == null) {
+                if (records.isEmpty()) {
+                    reportUntracedLeak(resourceType);
+                } else {
+                    reportTracedLeak(resourceType, records);
+                }
+            }
+        }
+        if (!reportedLeaks.keySet().isEmpty()) {
```

Comment:
Store the `Set` before calling `isEmpty()` a you also access it in the for loop.